packages/block-library/src/template-part/index.php

```diff
@@ -0,0 +1,60 @@
+<?php
+/**
+ * Server-side rendering of the `core/template-part` block.
+ *
+ * @package WordPress
+ */
+
+/**
+ * Renders the `core/template-part` block on the server.
+ *
+ * @param array $attributes The block attributes.
+ *
+ * @return string The render.
+ */
+function render_block_core_template_part( $attributes ) {
+	$content = null;
+
+	if ( ! empty( $attributes['postId'] ) ) {
+		// If we have a post ID, which means this template part
+		// is user-customized, render the corresponding post content.
+		$content = get_post( $attributes['postId'] )->post_content;
+	} elseif ( wp_get_theme()->get( 'TextDomain' ) === $attributes['theme'] ) {
+		// Else, if the template part was provided by the active theme,
+		// render the corresponding file content.
+		$template_part_file_path =
+				get_stylesheet_directory() . '/block-template-parts/' . $attributes['slug'] . '.html';
+		if ( file_exists( $template_part_file_path ) ) {
+			$content = file_get_contents( $template_part_file_path );
+		}
+	}
```

Comment:
I'm pretty sure the logic here might need to evolve as we discover edge cases and we polish the flows. I'm not certain the current way of handing theme template parts is the best, but it's a decent one to start with.

Reply:
We'll need to handle recursive template parts differently, maybe.